### PR TITLE
[Silabs] Update WiseConnect-wifi-bt-sdk to 2.10.3 for RS9116

### PR DESCRIPTION
**Pull request by bhmanda-silabs**
**Change**

- Updated WiseConnect-wifi-bt-sdk to 2.10.3 
- Added wpa_wpa2_mixed security type in rsi_if.cpp

**Testing**
Validated commissioning with mg24 + RS9116 combination

### CODE REVIEW

**Review by mkardous-silabs — COMMENTED** [review]
This requires a docker update in a seperate PR before hand otherwise the CI will never pass.

**Comment by bhmanda-silabs** [comment]
> This requires a docker update in a seperate PR before hand otherwise the CI will never pass.

Done @mkardous-silabs.

File: .github/workflows/release_artifacts.yaml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
         runs-on: ubuntu-latest
 
         container:
-            image: ghcr.io/project-chip/chip-build-esp32:81
```
Comment (mkardous-silabs) [inline .github/workflows/release_artifacts.yaml:35]: I don't think we should be updating the esp docker image.

**Comment by mkardous-silabs** [comment]
#36086 needs to be merged for this PR to be able to pass the necessary CIs.

File: integrations/cloudbuild/smoke-test.yaml
Comment (mkardous-silabs) [inline integrations/cloudbuild/smoke-test.yaml:1]: Does this file need to be updated?

**Review by mkardous-silabs — APPROVED** [review]

File: .github/workflows/release_artifacts.yaml (hunk 1 above)
Comment (bhmanda-silabs) [reply .github/workflows/release_artifacts.yaml]: Updated @mkardous-silabs

**Review by jmartinez-silabs — APPROVED** [review]

**Review by yufengwangca — APPROVED** [review]

**Pull request merged**